### PR TITLE
[DJFS-112] Develop a List View for Customer Orders And Store The Data in A Cookie

### PR DESCRIPTION
**Pull request by BenjaminKamena**
## Task: `Orders query file`
- `added the orders query file`

### CODE REVIEW

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+include('connect.php');
+
+if(isset($_GET['customer_id'])){
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:4]: So, this is what you had previously 
image

How is it possible for you to have a GET request that contains `customer_id`. Remember the question said then click on the `View` button, it should show is this page.

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ $customer_id =  $_GET['customer_id'];
+}
+$orders = isset($_COOKIE['orders'])? unserialize($_COOKIE['orders']) : $queryDatabase->getAllOrders();
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:7]: Also, I can't find the query you are using to get `$orders`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/benjaminKamena/phpDataase/queryDb.php [hunk 3]
```diff
@@ -19,7 +19,8 @@
 		echo "<td>" .$row['customer_first_name']. " " .$row['customer_last_name']. "</td>";
 		echo "<td>" .$row['customer_email_address']. "</td>";
 		echo "<td>" .$row['created_at']. "</td>";
-		echo "<td>" . '<button><a href ="#"> Veiw </a></button>'."</td>";
+		echo "<td>" . '<button>View</button>' . "</td>";
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/queryDb.php:22]: Remove this View button

File: submissions/benjaminKamena/phpDataase/queryDb.php [hunk 4]
```diff
@@ -19,7 +19,8 @@
...
+		echo "<td>" . '<button><a href ="./ordersQueryDb.php?id='. $row['id'] .'"> Veiw </a></button>'."</td>";
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/queryDb.php:23]: Typo in the spelling of View

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    LEFT JOIN orders ON orders.id = order_items.order_id
+    LEFT JOIN products ON products.id = order_items.product_id
+    LEFT JOIN addresses ON orders.customer_orders = addresses.customer_id
+    WHERE orders.customer_orders = '$id'";
+
+    $stmt = $this->conn()->query($sql);
+    $orders = $stmt->fetchAll();
+	setcookie("view", serialize($orders), time()+7200, "/");
+	
+	$orders = isset($_COOKIE['view']) ? unserialize($_COOKIE['view']) : $view->getAllCustomers($id);
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:19]: The way you wrote your code, anytime there is a page reload, the `SELECT` query will run even when cookie is set. This isn't the idea. What we want is that the `SELECT` query only run when cookie has expired.

A better implementation is to separate your query from your view (I,e what is seen on the browser). You can put the query is something like a function. Then only invoke or call this function when cookie is not set

**Review by adaugochi — CHANGES_REQUESTED** [review]
Well done so far

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	$sql = "SELECT order_items.quantity, order_items.unit_price, 
+            order_items.total_amount, orders.order_date, orders.id,customers.street_address, customers.city,customers.state, 
+            customers.country,products.name 
+            FROM order_items 
+            LEFT JOIN orders ON orders.id = order_items.order_id
+            LEFT JOIN customers on orders.customer_id = customers.id
+            LEFT JOIN products ON products.id = order_items.product_id 
+            WHERE orders.customer_id = '$id' ";
+            $stmt = $pdo->query($sql);
+            $orders = $stmt->fetchAll();
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:18]: Kindly separate your query from your view. Just the way you did for this previous task,

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            setcookie("orders",  serialize($orders), time() + 3600 );
+			} catch (PDOException $e) {
+				die("ERROR: Could not execute $sql. " . $e->getMessage());
+            }
+        };
+if(isset($_COOKIE['orders'])){
+    $orders =  unserialize($_COOKIE['orders']);
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:24]: What if the cookie is not set, where will the `$orders` come from?

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            LEFT JOIN customers on orders.customer_id = customers.id
+            LEFT JOIN products ON products.id = order_items.product_id 
+            WHERE orders.customer_id = '$id' ";
+            $stmt = $pdo->query($sql);
+            $orders = $stmt->fetchAll();
+            setcookie("orders",  serialize($orders), time() + 3600 );
+			} catch (PDOException $e) {
+				die("ERROR: Could not execute $sql. " . $e->getMessage());
+            }
+        };
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:22]: Kindly format your `catch` block properly

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+if(isset($_COOKIE['orders'])){
+    $orders =  unserialize($_COOKIE['orders']);
+    try{
+        if (count($orders) > 0) {
+            echo "<table>";
+			echo "<tr>";
+			echo "<td>ID</td>";
+			echo "<td>Product Name</td>";
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:30]: fromat line 28 - 30 properly

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+            setcookie("orders",  serialize($orders), time() + 3600 );
+        }catch (PDOException $e) {
+            die("ERROR: Could not execute $sql. " . $e->getMessage());
+        }
+    };
+
+    if(isset($_COOKIE['orders'])){
+        $orders =  unserialize($_COOKIE['orders']);
+} else{
+	$new_data = $order -> view($id);
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:27]: I don't understand this line of code. I mean on line 27

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by BenjaminKamena** [comment]
i was answering this question :What if the cookie is not set, where will
the $orders come from?

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+            $orders = $stmt->fetchAll();
+            setcookie("orders",  serialize($orders), time() + 3600 );
+        }catch (PDOException $e) {
+            die("ERROR: Could not execute $sql. " . $e->getMessage());
+        }
+    };
+
+    if(isset($_COOKIE['orders'])){
+        $orders =  unserialize($_COOKIE['orders']);
+    }
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:26]: We are making progress. But, If the cookie is not set, what should happen?

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by BenjaminKamena** [comment]
I made the changes

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php (hunk 11 above)
Comment (BenjaminKamena) [reply submissions/benjaminKamena/phpDataase/ordersQueryDb.php:28]: Am using cookies from previous task

Comment (adaugochi) [reply submissions/benjaminKamena/phpDataase/ordersQueryDb.php:28]: I don't get you. But, my question is if the cookie `orders` is not  where will the `$orders` be coming from?

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 12]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+include('connected.php');
+$id =  $_GET['id'];
+
+function view($id){
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:6]: This function does not have a closing `}`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+        }catch (PDOException $e) {
+            die("ERROR: Could not execute $sql. " . $e->getMessage());
+        }
+    };
+
+    if(isset($_COOKIE['orders'])){
+        $orders = isset($_COOKIE['orders'])? unserialize($_COOKIE['orders']) : $queryDatabase->getAllCustomers();
+    }
+    try{
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:29]: Move this action outside of this functio

File: submissions/benjaminKamena/phpDataase/ordersQueryDb.php [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+
+            setcookie("orders",  serialize($orders), time() + 3600 );
+
+        }catch (PDOException $e) {
+            die("ERROR: Could not execute $sql. " . $e->getMessage());
+        }
+    };
+
+    if(isset($_COOKIE['orders'])){
+        $orders = isset($_COOKIE['orders'])? unserialize($_COOKIE['orders']) : $queryDatabase->getAllCustomers();
```
Comment (adaugochi) [inline submissions/benjaminKamena/phpDataase/ordersQueryDb.php:27]: Where is this `$queryDatabase->getAllCustomers()` coming from?

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — APPROVED** [review]

**Pull request merged**